packages/patternfly-4/react-core/src/components/Avatar/Avatar.tsx

```diff
@@ -1,8 +1,6 @@
-
 import styles from '@patternfly/patternfly/components/Avatar/avatar.css';
 import { css } from '@patternfly/react-styles';
-import React, { DetailedHTMLProps, ImgHTMLAttributes } from 'react';
```

Comment:
Can we use the import { name, name } from 'react' syntax instead of * as React please?  We might need to add imports to this line instead of switching to the line below.

Reply:
if we remove the` import * as React` it would compile to:
```
import { ReactNode } from 'react';

const foo: ReactNode = <div />
```
for example... and in this case:
```
const foo = React.createElement('div') //React is undefined
```

Reply:
Seems that React is currently only exported as a default export

Reply:
https://unpkg.com/react@16.6.3/index.js

Reply:
For some more background on this, it all goes back to how babel and TS decided to handle commonjs modules interop with ES Module syntax. React exports a commonJS module (as seen in the link above). They do this instead of an ESModule since they support running React in a Node.js context. Example

Given the following file:
```js
function foo() {}
module.exports = foo;
```
You can import this with babel doing the following
```js
import foo from './foo'
```
With TypeScript Pre 2.7 you had to import the module doing the following
```ts
import * as foo from './foo';
```
Node finally decided that when importing a commonJS module from an ES Module you can import it using the same syntax babel decided on. With TypeScript 2.7+ `esModuleInterop` was added to match this behavior. 

This is great for applications since they can now have standard code on server and client. For libraries though this poses a problem since it requires any consumers of a library to also set `allowSyntheticDefaultImports` for the typings to be correct.

According to the React source and the React typings React does not have a `default` export. They only export a namespace. So if PF emits typing files that have `import React from 'react'` things will fail until a consumer sets `allowSyntheticDefaultImports` to `true`. On the flip side if PF exports a typing with `import * as React from 'react'` then both consumers using esModuleInterop and not using it will work. 

As for the question above for can we use the syntax of `import { FunctionComponent, ImgHTMLAttributes } from 'react'; That is not possible. For React to work `React` (the variable) has to be within the module (unless configured to consume from a CDN and it is a global var). This is the same for any project using babel or typescript. Anything needed off of the React namespace can be referenced from `React.FunctionComponent` instead of importing it. There are ways around this by setting `jsxFactory` in tsconfig of `jsxPragma` in babel config, but it is very much nonstandard and would likely require additional lint or IDE configuration.

If anyone is more curious about this I definitely recommend messing around with the [babel repl](https://babeljs.io/en/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4AoctCAOwGcsII4BeOAHgBNgA3OAegB8pIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.3.4) to view what is actually output when running through a compiler. The best and most up to date documentation is always source code.

Reply:
Further reading in this React [issue](https://github.com/facebook/react/issues/11503) would seem to support this.